src/components/backend-ai-console.ts

```diff
@@ -399,13 +400,21 @@ export default class BackendAIConsole extends connect(store)(LitElement) {
     });
     this.addTooltips();
     this.sidebarMenu.style.minHeight = (this.is_admin || this.is_superadmin) ? '600px' : '250px';
-    if (!this.is_admin || !this.is_superadmin) {
+    // redirect to unauthorized page when user's role is neither admin nor superadmin
+    if (!this.is_admin && !this.is_superadmin) {
       if (this.adminOnlyPages.includes(this._page) || this._page === 'unauthorized') {
         this._page = 'unauthorized';
         globalThis.history.pushState({}, '', '/unauthorized');
         store.dispatch(navigate(decodeURIComponent(this._page)));
       }
     }
+
+    // redirect to unauthorize page when admin user tries to access superadmin only page
+    if (this.is_admin && this.superAdminOnlyPages.includes(this._page)) {
```

Comment:
`this.is_admin` is`true` for superadmin, either. This `if` statement makes superadmins redirect to unauthorized page when accessing `superAdminOnlyPages` by typing address directly. For example, login as a superadmin and type in the browser URL `http://<test-server-ip:port>/agent`.